src/plugins/astro/index.ts

```diff
@@ -0,0 +1,17 @@
+import { hasDependency } from '../../util/plugin.js';
+import type { IsPluginEnabledCallback } from '../../types/plugins.js';
+
+// https://docs.astro.build/en/reference/configuration-reference/
+
+export const NAME = 'astro';
+
+/** @public */
+export const ENABLERS = ['astro'];
+
+export const isEnabled: IsPluginEnabledCallback = ({ dependencies }) => hasDependency(dependencies, ENABLERS);
+
+export const ENTRY_FILE_PATTERNS = ['astro.config.{js,cjs,mjs,ts}', 'src/content/config.ts'];
+
+export const PRODUCTION_ENTRY_FILE_PATTERNS = ['src/pages/**/*.{astro,md,mdx,html,js,ts}'];
```

Comment:
I think the `.md` and `.html` files can be ignored, as they don't `import` anything? Maybe `.html` could be "compiled", but could be tricky to find `import` statements only within `<script>` tags? I'd be fine to leave that out for now.

There can also be `.mdx` files in `src/content` dir, so this would be my suggestion:

```suggestion
export const PRODUCTION_ENTRY_FILE_PATTERNS = ['src/pages/**/*.{astro,mdx,js,ts}', 'src/content/**/*.mdx'];
```

Reply:
Good catch! I believe for imports to be "resolved" by astro, the extension has to be .astro anyways, so .html is just copied: https://docs.astro.build/en/core-concepts/astro-components/#html-components - and I believe same is true for .md, just translated into .html